server/controllers/notification.js

```diff
@@ -0,0 +1,81 @@
+const asyncHandler = require("express-async-handler");
+const { PriceDropNotification } = require("../models/Notification");
+const Product = require("../models/Product");
+
+exports.createNotification = asyncHandler(async (req, res) => {
+  const { userId } = req.params;
+  const { productId } = req.body;
+  const alreadyAdded = await PriceDropNotification.find({ productId, userId });
+  if (alreadyAdded) {
+    res.status(409).send({ message: "Notification already Added" });
+  }
+  const product = await Product.findById(productId);
+  if (!product) {
+    res.status(404).send({ message: "Product not Found" });
+  } else {
+    const priceDropNotification = await PriceDropNotification.create({
+      productId,
+      oldPrice: product.price,
+      url: product.url,
+      read: false,
+      userId,
+    });
+
+    if (priceDropNotification) {
+      res.status(201).json({
+        success: {
+          priceDropNotification: {
+            id: priceDropNotification._id,
+            productId: priceDropNotification.productId,
+            oldPrice: priceDropNotification.oldPrice,
+            url: priceDropNotification.url,
+            userId: priceDropNotification.userId,
+          },
+        },
+      });
+    } else {
+      res.status(400);
+      throw new Error("Invalid notification data");
```

Comment:
throwing an error without a message sent back as a response seems inconsistent with other routes

Reply:
Right. Corrected. Thank you for noting.